src/utils/__tests__/removeUnnecessarySpaces.test.ts

```diff
@@ -0,0 +1,16 @@
+import { removeUnnecessarySpaces } from "../removeUnnecessarySpaces";
+
+describe("removeUnnecessarySpaces", () => {
+  test("should return string without unnecessary spaces", () => {
+    const query = `       query Me {\n viewer     { id name
+      email }
+    }`;
+
+    expect(removeUnnecessarySpaces(query))
+      .toBe("query Me { viewer { id name email } }");
+  });
+
+  test.each([undefined, null, "", 0, true, false, {}, []])("wrong value: %p", (payload) => {
+    expect(removeUnnecessarySpaces(payload as never)).toBe("");
+  });
```

Comment:
`removeUnnecessarySpaces` only takes a `string` or `undefined`. Assuming we have strict typings, these edgecases seem unnecessary. It may also be masking issues if we pass in an `object` or an `array` and get a blank string back.

Reply:
removed unnecessary code